R-packages/evalcast/R/scale_by_forecaster.R

```diff
@@ -0,0 +1,61 @@
+#' Scale error measures based on those of a particular forecaster.
+#'
+#' @param score_card score_card like that returned by `evaluate_predictions()`
+#' @param score_cols vector of column names in `score_card` to normalize
+#' @param base_forecaster_name name of forecaster in `score_card$forecaster` column by whose error
+#'   values the remaining forecasters' errors will be scaled
+#' @param numeric_err_cols vector of column names in `score_card` that contain numeric error
+#'   measures. Elements of `numeric_err_cols` that are not in `score_cols` will be dropped from the
+#'   final output to avoid a mix of scaled and unscaled measures.
+#'
+#' @return
+#' @export
+scale_by_forecaster <- function(score_card,
+                                score_cols,
+                                base_forecaster_name, 
+                                numeric_err_cols = c("ae", "wis")) {
+    # Validate columns arguments
+    columns <- colnames(score_card)
+    assert_that("forecaster" %in% columns,
+                msg = 'score_card must have a column named "forecaster"')
+    
+    extraneous_score_cols <- setdiff(score_cols, columns)
+    assert_that(length(extraneous_score_cols) == 0,
+                msg = paste("score_cols contains columns",
+                            paste(extraneous_score_cols, collapse=", "),
+                            "not present in the columns of score_card"))
+
+    nonnumeric_score_cols <- setdiff(score_cols, numeric_err_cols)
+    assert_that(length(nonnumeric_score_cols) == 0,
+                msg = paste("score_cols contains columns",
+                            paste(nonnumeric_score_cols, collapse=", "),
+                            "that are not numeric errors"))
+
```

Comment:
Some of this will become unnecessary given my proposed changes to the arguments.